ast/src/main/java/com/graphicsfuzz/common/tool/PrettyPrinterVisitor.java

```diff
@@ -572,10 +572,34 @@ public void visitTernaryExpr(TernaryExpr ternaryExpr) {
 
   @Override
   public void visitArrayIndexExpr(ArrayIndexExpr arrayIndexExpr) {
-    visit(arrayIndexExpr.getArray());
-    out.append("[");
-    visit(arrayIndexExpr.getIndex());
-    out.append("]");
+    /* If arrayIndexExpr indexes into a literal-to-uniform array, output a #define instead. */
+    if (arrayIndexExpr.getArray().getText().contains(Constants.INT_LITERAL_UNIFORM_VALUES)
```

Comment:
Instead of asking whether the text of the arrayIndexExpr contains the string, check that the expression is a variable identifier expression and that, if so, it's name is .equals to the string:

```
if (arrayIndexExpr instanceof VariableIdentiferExpr && ((VariableIdentifierExpr) arrayIndexExpr).getName().equals(...)
```

What you did with checking substrings might work in practice but it's a bit fragile.

Reply:
Changed.